<a href="https://colab.research.google.com/github/pragnakulkarni/Netflix-Movies-and-TV-Shows/blob/main/Welcome_To_Colab.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [9]:
!wget https://raw.githubusercontent.com/pragnakulkarni/Netflix-Movies-and-TV-Shows/main/netflix_titles.csv

--2024-07-26 08:42:01--  https://raw.githubusercontent.com/pragnakulkarni/Netflix-Movies-and-TV-Shows/main/netflix_titles.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.110.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 3399671 (3.2M) [text/plain]
Saving to: ‘netflix_titles.csv.1’

netflix_titles.csv. 100%[===================>]   3.24M  --.-KB/s    in 0.08s   

2024-07-26 08:42:01 (39.1 MB/s) - ‘netflix_titles.csv.1’ saved [3399671/3399671]



Let's import the required libraries

In [10]:
import pandas as pd
import numpy as np
import seaborn as sns

Let's create a Pandas DataFrame for the dataset

In [11]:
netflix_df = pd.read_csv("/content/netflix_titles.csv")
netflix_df

show_id     type                  title         director  \
0         s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1         s2  TV Show          Blood & Water              NaN   
2         s3  TV Show              Ganglands  Julien Leclercq   
3         s4  TV Show  Jailbirds New Orleans              NaN   
4         s5  TV Show           Kota Factory              NaN   
...      ...      ...                    ...              ...   
8802   s8803    Movie                 Zodiac    David Fincher   
8803   s8804  TV Show            Zombie Dumb              NaN   
8804   s8805    Movie             Zombieland  Ruben Fleischer   
8805   s8806    Movie                   Zoom     Peter Hewitt   
8806   s8807    Movie                 Zubaan      Mozez Singh   

                                                   cast        country  \
0                                                   NaN  United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2     Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                   NaN            NaN   
4     Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   
...                                                 ...            ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8803                                                NaN            NaN   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...          India   

              date_added  release_year rating   duration  \
0     September 25, 2021          2020  PG-13     90 min   
1     September 24, 2021          2021  TV-MA  2 Seasons   
2     September 24, 2021          2021  TV-MA   1 Season   
3     September 24, 2021          2021  TV-MA   1 Season   
4     September 24, 2021          2021  TV-MA  2 Seasons   
...                  ...           ...    ...        ...   
8802   November 20, 2019          2007      R    158 min   
8803        July 1, 2019          2018  TV-Y7  2 Seasons   
8804    November 1, 2019          2009      R     88 min   
8805    January 11, 2020          2006     PG     88 min   
8806       March 2, 2019          2015  TV-14    111 min   

                                              listed_in  \
0                                         Documentaries   
1       International TV Shows, TV Dramas, TV Mysteries   
2     Crime TV Shows, International TV Shows, TV Act...   
3                                Docuseries, Reality TV   
4     International TV Shows, Romantic TV Shows, TV ...   
...                                                 ...   
8802                     Cult Movies, Dramas, Thrillers   
8803             Kids' TV, Korean TV Shows, TV Comedies   
8804                            Comedies, Horror Movies   
8805                 Children & Family Movies, Comedies   
8806     Dramas, International Movies, Music & Musicals   

                                            description  
0     As her father nears the end of his life, filmm...  
1     After crossing paths at a party, a Cape Town t...  
2     To protect his family from a powerful drug lor...  
3     Feuds, flirtations and toilet talk go down amo...  
4     In a city of coaching centers known to train I...  
...                                                 ...  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...  

[8807 rows x 12 columns]

Datatypes of the dataframe and the individual column

In [13]:
netflix_df.dtypes
netflix_df['date_added'].dtype

dtype('O')

Let's see the number of rows and columns using the shape() function

In [15]:
netflix_df.shape

(8807, 12)

**8807** rows and **12** columns

Let's see for any null values present in the dataframe

In [16]:
netflix_df.isnull().sum()

show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64

Dealing with null values in the "director" column

In [28]:
# Let's see how much percentage of director column has null values
missing_proportion = netflix_df['director'].isna().mean() * 100
'''missing_proportion = 29.908027705234474
# since the missing proportion is approximately 30% of the column,
#  so we are not going to drop those null values and are replacing them with the "Unknown" value'''
netflix_df['director'].fillna("Unknown", inplace = True)

Dealing with null values in the cast column

In [39]:
# Let's see how much percentage of cast column has null values
cast_missing = netflix_df['cast'].isnull().mean() * 100
# cast_missing = 9.367548540933349
''' since the missing proportion is approximately 10% of the column,
we are not going to drop those null values and are replacing them with the "Unknown" value'''
netflix_df['cast'].replace(np.NaN,'Unknown', inplace = True)

Dealing with null values in the country column

In [43]:
netflix_df['country'].isnull().sum()

831